### PR TITLE
gluon-web-osm: add javascript static osm source

### PR DESCRIPTION
**Pull request by 2tata**
This package provides a static source for osm.js, which can be use to show an Open Street Map map on the status-page or in the config-mode.

As I discuss with @NeoRaider XMPP. I put the osm.js source in an extra package. To use this source developers need add to their packages Makefile just a symlink e.g.

`$(LN) /lib/gluon/web/www/static/osm.js $(1)/lib/gluon/config-mode/www/static/`

This package is spitted up from #1211 

Signed-off-by: Jan-Tarek Butt <tarek@ring0.de>

### CODE REVIEW

File: package/gluon-web-osm/javascript/osm.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                                oLon = findObj("longitude");
+                                oLat = findObj("latitude");
+                                lonlat1 = new OpenLayers.LonLat(i.lon, i.lat).transform(a, e),
+                                        oLon.value = lonlat1.lon,
+                                        oLat.value = lonlat1.lat,
+                                        n.clearMarkers(),
+                                        n.addMarker(new OpenLayers.Marker(i)),
+                                        oLon.className = oLon.className.replace(/ gluon-input-invalid/g, ""),
+                                        oLat.className = oLat.className.replace(/ gluon-input-invalid/g, "");
+                                        //Anyone knows how to trigger the change event?
```
Comment (rotanid) [inline package/gluon-web-osm/javascript/osm.js]: we don't merge code with comments/placeholders like these ;)
if https://www.w3schools.com/jsref/event_onchange.asp doesnt help (or googling) remove these lines please.

**Comment by 2tata** [comment]
On 07/25/18 02:51, Andreas Ziegler wrote:
> rotanid commented on this pull request.
> 
> 
> 
>> +                                this.handlerOptions = OpenLayers.Util.extend({}, this.defaultHandlerOptions), OpenLayers.Control.prototype.initialize.apply(this, arguments), this.handler = new OpenLayers.Handler.Click(this, {
> +                                        click: this.trigger
> +                                }, this.handlerOptions)
> +                        },
> +                        trigger: function(t) {
> +                                var i = osmMap.getLonLatFromPixel(t.xy);
> +                                oLon = findObj("longitude");
> +                                oLat = findObj("latitude");
> +                                lonlat1 = new OpenLayers.LonLat(i.lon, i.lat).transform(a, e),
> +                                        oLon.value = lonlat1.lon,
> +                                        oLat.value = lonlat1.lat,
> +                                        n.clearMarkers(),
> +                                        n.addMarker(new OpenLayers.Marker(i)),
> +                                        oLon.className = oLon.className.replace(/ gluon-input-invalid/g, ""),
> +                                        oLat.className = oLat.className.replace(/ gluon-input-invalid/g, "");
> +                                        //Anyone knows how to trigger the change event?
> 
> we don't merge code with comments/placeholders like these ;)
> if https://www.w3schools.com/jsref/event_onchange.asp doesnt help (or googling) remove these lines please.
> 

Oh yes, will remove it. This comment is old and out of the source base which I 
have used.

**Comment by rotanid** [comment]
@adlerweb as most of this code is from your repo at https://github.com/FreifunkMYK/gluon-packages-extra/tree/master/gluon-config-mode-geo-location-map
are you OK with Gluon merging this code under the BSD 2 clause license?

**Comment by adlerweb** [comment]
fine by me

**Pull request merged**